### DISCUSSION TITLE
How to reset the password for Mirth Connect (3.12)?

### DISCUSSION THREAD

**Discussion by Coooz** [nextgenhealthcare/connect#6176] [Q&A]
I need to upgrade Mirth Connect with a customer of ours from 3.12.0 to 4.5.0. It appears that with this customer we no longer have the login credentials for Mirth Connect, so I've been looking for a way to reset the admin password. This was the most promising result:
https://forums.mirthproject.io/forum/mirth-connect/support/13131-incorrect-username-or-password-after-reboot-3-x.
However, version 3.03 is a long way from version 3.12 - I don't know whether this still works and I don't want to mess things up. Can I still follow the procedure that is outlined here?

Thanks in advance!

**Comment by jonbartels** [comment]
A safer path is to create a NEW user in the DB then set that users password to `admin`. 

A search for "person_password" on Github and the forums would help locate more recent threads with the latest instructions. 

A landmine to watch for is that the password hashing logic changed, I think in the later 3.x releases.I think it updates the hashes on the first login from a user with the old-style hash so INSERTing rows with the older pattern should still work.

What database engine are you using?

One way to recreate a user with the password `admin` is to review and replay the initial user creation scripts. For example postgres creates the new user here: https://github.com/nextgenhealthcare/connect/blob/c1bbf981b396279984124adbd50ddb4851b2bc76/server/dbconf/postgres/postgres-database.sql

**Reply by Coooz** [reply]
Thank you, @jonbartels , this looks like the path to follow indeed - and of course I can always restore a backup of the database if things don't work out as what I hoped for. So that's safe. I use SQL Server by the way.
